libminifi/test/aws-tests/FetchS3ObjectTests.cpp

```diff
@@ -0,0 +1,174 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <fstream>
+
+#include "S3TestsFixture.h"
+#include "processors/FetchS3Object.h"
+
+class FetchS3ObjectTestsFixture : public S3TestsFixture<minifi::aws::processors::FetchS3Object> {
+ public:
+  FetchS3ObjectTestsFixture() {
+    auto putfile = plan->addProcessor(
+      "PutFile",
+      "PutFile",
+      core::Relationship("success", "d"),
+      true);
+    char output_dir_mask[] = "/tmp/gt.XXXXXX";
+    output_dir = test_controller.createTempDirectory(output_dir_mask);
+    plan->setProperty(putfile, "Directory", output_dir);
+  }
+
+  std::string output_dir;
+};
+
+TEST_CASE_METHOD(FetchS3ObjectTestsFixture, "Test AWS credential setting", "[awsCredentials]") {
+  setBucket();
+
+  SECTION("Test property credentials") {
+    setAccesKeyCredentialsInProcessor();
+  }
+
+  SECTION("Test credentials setting from AWS Credentials service") {
+    setAccessKeyCredentialsInController();
+    setCredentialsService();
+  }
+
+  SECTION("Test credentials file setting") {
+    setCredentialFile(s3_processor);
+  }
+
+  SECTION("Test credentials file setting from AWS Credentials service") {
+    setCredentialFile(aws_credentials_service);
+    setCredentialsService();
+  }
+
+  SECTION("Test credentials setting using default credential chain") {
+    setUseDefaultCredentialsChain(s3_processor);
+  }
+
+  SECTION("Test credentials setting from AWS Credentials service using default credential chain") {
+    setUseDefaultCredentialsChain(aws_credentials_service);
+    setCredentialsService();
+  }
+
+  test_controller.runSession(plan, true);
+  REQUIRE(mock_s3_wrapper_ptr->getCredentials().GetAWSAccessKeyId() == "key");
+  REQUIRE(mock_s3_wrapper_ptr->getCredentials().GetAWSSecretKey() == "secret");
+}
+
+TEST_CASE_METHOD(FetchS3ObjectTestsFixture, "Test required property not set", "[awsS3Config]") {
+  SECTION("Test credentials not set") {
+  }
+
+  SECTION("Test no bucket is set") {
+    setAccesKeyCredentialsInProcessor();
+  }
+
+  SECTION("Test no object key is set") {
+    setRequiredProperties();
+    plan->setProperty(update_attribute, "filename", "", true);
+  }
+
+  SECTION("Test region is empty") {
+    setRequiredProperties();
+    plan->setProperty(s3_processor, "Region", "");
+  }
+
+  REQUIRE_THROWS_AS(test_controller.runSession(plan, true), minifi::Exception);
+}
+
+TEST_CASE_METHOD(FetchS3ObjectTestsFixture, "Test proxy setting", "[awsS3Proxy]") {
+  setRequiredProperties();
+  setProxy();
+  test_controller.runSession(plan, true);
+  checkProxySettings();
+}
+
+TEST_CASE_METHOD(FetchS3ObjectTestsFixture, "Test default properties", "[awsS3Config]") {
+  setRequiredProperties();
+  test_controller.runSession(plan, true);
+  REQUIRE(LogTestController::getInstance().contains("key:s3.bucket value:" + S3_BUCKET));
```

Comment:
Minor but `contains` is slow. `utils::verifyLogLinePresenceInPollTime` should be a better choice here.

Reply:
Replaced it in [840a645](https://github.com/apache/nifi-minifi-cpp/pull/970/commits/840a645918fb63a6bde7a8c702e0a58125f0f085).